src/main/java/nextstep/subway/path/application/PathService.java

```diff
@@ -0,0 +1,22 @@
+package nextstep.subway.path.application;
+
+import lombok.RequiredArgsConstructor;
+import nextstep.subway.path.domain.Path;
+import nextstep.subway.path.domain.SubwayGraph;
+import nextstep.subway.station.application.StationReader;
+import nextstep.subway.station.domain.Station;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class PathService {
+  private final GraphService graphService;
+  private final StationReader stationReader;
+
+  public Path findPath(PathRequest request) {
+    SubwayGraph graph = graphService.loadGraph();
+    Station source = stationReader.readById(request.getSource());
+    Station sink = stationReader.readById(request.getTarget());
+    return graph.getShortestPath(source, sink);
```

Comment:
만약 source, sink Station이 유효하지 않은 Station이라면 
`SubwayGraph graph = graphService.loadGraph();` 을 수행하지 않고 에러 응답을 해줄 수 있을 것 같아요.

Reply:
```java
@Component
@RequiredArgsConstructor
public class StationReader {
  private final StationRepository stationRepository;

  public Station readById(Long id) {
    return stationRepository.findById(id).orElseThrow(() -> new StationNotFoundException(id));
  }
}
```

에서 `Station`이 유효하지 않으면 예외를 던지고

```java
@RestControllerAdvice
public class ApiControllerAdvice {
  private final Logger logger = LoggerFactory.getLogger(getClass());

  @ExceptionHandler(ApiException.class)
  public ResponseEntity<ErrorResponse> handleApiException(ApiException e) {
    logger.warn("ApiException : {}", e.getMessage(), e);
    return new ResponseEntity<>(
        ErrorResponse.of(e.getErrorCode(), e.getData()), e.getErrorCode().getStatus());
  }

그 외...
}
```
에서 예러 HTTP 응답을 하도록 구현했습니다.

Reply:
```suggestion
    Station source = stationReader.readById(request.getSource());
    Station sink = stationReader.readById(request.getTarget());
    SubwayGraph graph = graphService.loadGraph();
    return graph.getShortestPath(source, sink);
```
저는 `SubwayGraph graph = graphService.loadGraph();`의 호출 위치를 Station을 조회하는데 밑에 두게 되면 예외처리가 되는 상황에서 불필요한 loadGraph() 행위를 하지 않게끔 할 수 있지 않을까? 하는 의견을 드렸던 것이 었습니다 ㅎㅎ 제가 모호하게 말씀드렸군요. 🥲